### ISSUE TITLE
Docker for Windows fails to start with error Binary stream '0' does not contain a valid BinaryHeader. Possible causes are invalid stream or object version change between serialization and deserialization

### ISSUE THREAD

**Issue by plkumar** [docker/for-win#1355]
### Expected behavior
Docker for Windows does not start.

### Actual behavior

Binary stream '0' does not contain a valid BinaryHeader. Possible causes are invalid stream or object version change between serialization and deserialization.
   at Docker.Core.Pipe.NamedPipeClient.Send(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeClient.cs:line 36
   at Docker.Actions.DoStart(SynchronizationContext syncCtx, Boolean showWelcomeWindow, Boolean executeAfterStartCleanup) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 95
   at Docker.Actions.<>c__DisplayClass18_0.<Start>b__0() in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 81
   at Docker.WPF.TaskQueue.<>c__DisplayClass19_0.<.ctor>b__1() in C:\gopath\src\github.com\docker\pinata\win\src\Docker.WPF\TaskQueue.cs:line 59

[log.txt](https://github.com/docker/for-win/files/1499456/log.txt)

**Comment by HassanChehab** [comment]
I have exactly the same error. Hopefully, someone will address our issue.

**Comment by albamoro** [comment]
The link below worked for me
https://github.com/docker/for-win/issues/1232

**Comment by tonytronics** [comment]
Any update on this issue Im getting it on my docker installation

**Comment by rn** [comment]
do you have any AV software installed?

**Comment by AvneRedis** [comment]
This fix worked for me:
Had the same issue - I think that the root for this is the fact the I had VirtualBox installed on my machine. Tried the solution here #1232 but it didn't work. 
Eventaully I manually turned the Hyper-V feature in Windows - here's how you do it:

"Control Panel" > "Programs" > "Programs and Features" > "Turn Windows features on or off" 
Simply check the Hyper-V and press OK.

This will require you to reboot your machine.

Hope it works for you.
